src/processor/mod.rs

```diff
@@ -110,13 +110,14 @@ impl EventProcessor {
                 }
             },
             #[cfg(feature = "query")]
-            Message::Query(qry) => match qry.query.get_route() {
-                crate::query::QueryRoute::Log => {
-                    let pref = qry.query.get_query_data().data.i;
-                    println!("Respond with {} key event log.", pref.to_string());
+            Message::Query(qry) => match qry.query.event.content.data.route {
```

Comment:
Checking query signatures is missed here. We need similar logic like in `process_signed_query` in witness module. But maybe not here.
How about splitting `Message` into two types: those which save anything in db (and those should be processed here) and those which just read data (which can be handled somewhere else, maybe in witness `parse_and_process` function)?